src/plugins/intel_gpu/src/graph/impls/ocl/gemm.cpp

```diff
@@ -40,7 +40,9 @@ struct gemm_impl : typed_primitive_impl_ocl<gemm> {
             params.inputs.push_back(convert_data_tensor(input_layouts[i]));
         }
         params.outputs[0] = convert_data_tensor(output_layout);
-
+        if (impl_param.get_program().get_node(primitive->id).is_dynamic()) {
+            params.is_shape_agnostic = true;
```

Comment:
Why this flag is needed? You can always get the value inside kernel selector by checking if any of inputs/outputs tensor is dynamic

Reply:
In case of gemm_tiled_opt shape agnotic kernel, when `update_dispatch_data_func` is called, it is necessary to check whether the current selected kernel is static kernel or shape agnostic kernel in `SetTuningParams` of kernel selector. However, since the inputs/outputs tensor is already static, it cannot be checked. Therefore, additional flag is needed to determine this.

Reply:
What I don't like in current impl is that you've added is_shape_agnostic flag into common params, but fill it properly only for gemm primitive. So I see 2 possible options to solve it:
1. Determine if kernel is shape agnostic in gemm kernel by checking if tensors are dynamic, and then just capture the flag value in `update_dispatch_data_func`
2. Move initialization of this flag to `get_default_params`

The second option can be used if multiple primitives may require similar flags, otherwise the first option is preferrable.

Reply:
Yes we can move it to common param

Reply:
Because maybe FC is also expected to require the same flag, it's updated as second option.